src/Microsoft.Framework.Runtime.Hosting/Json/JsonContent.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            var content = new List<string>();
+            var reader = new StreamReader(stream);
+
+            string line = reader.ReadLine();
+            while (line != null)
+            {
+                content.Add(line);
+                line = reader.ReadLine();
+            }
+
+            return new JsonContent(content);
+        }
+
+        private JsonContent(List<string> content)
+        {
+            _content = content;
+        }
+
+        public int TotalLines { get { return _content.Count; } }
+
+        /// <summary>
+        /// Current line number in zero-based index.
+        /// </summary>
+        public int CurrentLine { get; private set; } = 0;
+
+        /// <summary>
+        /// Current column number in zero-based index.
+        /// </summary>
+        public int CurrentColumn { get; private set; } = -1;
+
+        public char CurrentChar
+        {
+            get { return _content[CurrentLine][CurrentColumn]; }
+        }
+
+        public bool ValidCursor
+        {
+            get
+            {
+                return CurrentLine < _content.Count &&
+                       CurrentLine >= 0 &&
+                       CurrentColumn < _content[CurrentLine].Length &&
+                       CurrentColumn >= 0;
+            }
+        }
+
+        public bool Started
+        {
+            get { return CurrentLine != 0 || CurrentColumn != -1; }
+        }
+
+        /// <summary>
+        /// Move the cursor to the next non empty char
+        /// </summary>
+        /// <returns>Returns false if the cursor reach the end of the content.</returns>
+        public bool MoveToNextNonEmptyChar()
+        {
+            while (TotalLines > CurrentLine)
+            {
+                while (_content[CurrentLine].Length > CurrentColumn + 1)
+                {
+                    char c = _content[CurrentLine][++CurrentColumn];
+                    if (!char.IsWhiteSpace(c))
+                    {
+                        return true;
+                    }
+                }
+
+                CurrentLine++;
+                CurrentColumn = -1;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Move the cursor to the next char
+        /// </summary>
+        /// <returns>Returns false if the cursor reach the end of the content.</returns>
+        public bool MoveNext()
+        {
+            if (CurrentColumn + 1 < _content[CurrentLine].Length)
+            {
+                CurrentColumn += 1;
+                return true;
+            }
+            else
+            {
+                // find the first non empty line after current line
+                var targetLine = CurrentLine;
+                while (++targetLine < TotalLines && _content[targetLine].Length == 0)
```

Comment:
I think it would be easier to understand if you move the increment inside the while block